### ISSUE TITLE
Prompt for BLIP2 OPT VQA

### ISSUE THREAD

**Issue by codezakh** [salesforce/LAVIS#138]
Hi @LiJunnan1992 and @dxli94, thanks for the great work! Do you have examples of the prompts you found to work well with the OPT version of BLIP2 for VQA? I tried using FLAN-T5's prompt, but it didn't work very well for the OPT models. Image captioning works fine, it's just question answering that I'm having trouble with.

**Comment by dxli94** [comment]
Hi @codezakh, we don't use text prompt for OPT captioning. OPT is a decoder LLM. During training, it is directly prompted with the query tokens. See Fig 3 top in the paper.

Thanks.

**Comment by joez17** [comment]
@codezakh What's your VQA result when using FLAN-T5's prompt on OPT version model? I tried similar prompt and find OPT version model hardly gives correct answer. It's confused.

**Comment by YuanLiuuuuuu** [comment]
Can you reproduce the results for OPT zeroshot VQA in the paper?

**Comment by codezakh** [comment]
Hi @YuanLiuuuuuu, this comment describes how they were able to reproduce the BLIP-2 OPT results. I haven't tried it yet: https://github.com/salesforce/LAVIS/issues/188

**Comment by YuanLiuuuuuu** [comment]
@codezakh Thanks!

**Comment by dszpr** [comment]
Hi！ @codezakh @YuanLiuuuuuu 
I noticed that in forward function in blip2_opt.py, only the questions in the VQA dateset are used. Both the text_input and the target lables are derived from the opt_tokens
`


        text = [t + "\n" for t in samples["text_input"]]

        opt_tokens = self.opt_tokenizer(
            text,
            return_tensors="pt",
            padding="longest",
            truncation=True,
            max_length=self.max_txt_len,
        ).to(image.device)

        targets = opt_tokens.input_ids.masked_fill(
            opt_tokens.input_ids == self.opt_tokenizer.pad_token_id, -100
        )
        if self.prompt:
            targets[:, : self.prompt_length] = -100  # do not apply loss to the prompt

        empty_targets = (
            torch.ones(atts_opt.size(), dtype=torch.long).to(image.device).fill_(-100)
        )
        targets = torch.cat([empty_targets, targets], dim=1)

        inputs_embeds = self.opt_model.model.decoder.embed_tokens(opt_tokens.input_ids)
        inputs_embeds = torch.cat([inputs_opt, inputs_embeds], dim=1)
        attention_mask = torch.cat([atts_opt, opt_tokens.attention_mask], dim=1)

        with self.maybe_autocast():
            outputs = self.opt_model(
                inputs_embeds=inputs_embeds,
                attention_mask=attention_mask,
                return_dict=True,
                labels=targets,
            )
`
In the VQA task,  are the target lables supposed to be the answers in the VQA dataset? But the answers in the VQA datasetare not used in the blip2_opt.py. However, the answers are used as target lables in blip2_t5.py. It really confused.
Have you make any change to blip2_opt.py?
